Assignment/palindrome_test.c

```diff
@@ -20,102 +20,101 @@
 #define RED_TEXT "\033[0;31m"
 #define RESET_TEXT "\033[0m"
 
-int IsValid (const char* str);
-void PrintRes (const char* input, const char* expectedOutput, int actualOutput);
-void PalindromeTest ();
-void ReverseNumPalindromeTest ();
+int IsValid (const char* num);
+void PrintRes (const char* inputs, const char* expOutputs, const char* actualRes, const char* res);
+void TestCases ();
 void UserInput ();
 
-int IsValid (const char* str) {
-   if (*str == '\0' || str == NULL) return 0;
-   if (*str == '-') str++;
-   while (*str) {
-      if (!isdigit ((unsigned char)*str)) return 0;
-      str++;
+/// <summary>To check if the input number is valid</summary>
+int IsValid (const char* num) {
+   if (*num == '\0' || num == NULL) return 0;
+   if (*num == '-') num++;
+   while (*num) {
+      if (!isdigit ((unsigned char)*num)) return 0;
+      num++;
    }
    return 1;
 }
 
-void PrintRes (const char* input, const char* expectedOutput, int actualOutput) {
-   const char* actualRes = actualOutput ? "Palindrome" : "Not a Palindrome";
-   const char* res = (strcmp (actualRes, expectedOutput) == 0) ? GREEN_TEXT "Pass" RESET_TEXT : RED_TEXT "Fail" RESET_TEXT;
-   printf ("| %-30s | %-20s | %-20s | %-27s |\n", input, expectedOutput, actualRes, res);
+void PrintRes (const char* inputs, const char* expOutputs, const char* actualRes, const char* res) { // Print the result

```

Comment:
Can remove this function as it is only used once and also it is a one-line statement. So, it can be directly written there.